### PR TITLE
Calling this finished

### PR DESCRIPTION
**Pull request by jsim0809**
This is my full basic component. It includes conditionally rendered components, modals, tooltips, and a scroll bar.

I've taken a look at some of my last few features and considered them shelved for now. These include:

- Mouseover and on-click tooltips for users (pretty elaborate, almost another component in itself, 1-2 days of work)
- Buttons changing color when they are hovered over (very simple, 15 minutes of work)
- Previous/Next buttons changing color and becoming unclickable when on the first/last page. (fairly simple, 30 minutes of work)
- Hiding the scroller when there is only 1 page of questions (fairly simple, 30 minutes of work)
- Sorting the questions by post date (easy, 15 minutes)
- Sorting the answers by upvotes and downvotes (easy, 15 minutes)
- Different box displaying when there are no answers to a question (30 minutes)
- 4 different layouts depending on screen size (2-4 hours)
- Cleaning up my code and improving my comments (1-2 hours)
- Improving my file structure for intuitiveness (1-2 hours)

I've decided to spend today on testing, because I think I'll learn much more from it. I may revisit these features over the weekend.

### CODE REVIEW

File: client/components/AnswerForm.jsx [hunk 1]
```diff
@@ -3,36 +3,21 @@ import GuidelinesModal from './GuidelinesModal';
...
+    {/* Posting guidelines -- pops up a modal. */}
+    <div className="guidelines float-right" onClick={handleGuidelinesClick}>Posting guidelines</div>
+    <div className="answer-form-buttonbar">
+      {/* Submit button -- pops up a fake login modal. */}
+      <button type="button" className="small-black-button" onClick={handleLoginClick}>Submit</button>
+      {/* Cancel button -- closes the form. */}
+      <button type="button" className="small-white-button" onClick={handleAnswerFormCancelClick}>Cancel</button>
+    </div>
+  </div>
+);
```
Comment (kcelentano) [inline client/components/AnswerForm.jsx]: A smart refactor from stateful to stateless! One thing, a blank line is needed at the end.

File: client/components/AskForm.jsx [hunk 2]
```diff
@@ -3,40 +3,24 @@ import GuidelinesModal from './GuidelinesModal';
...
-      </div>
-    );
-  }
-}
+    {/* Cancel button -- closes the form. */}
+    <button type="button" className="large-white-button" onClick={handleAskFormCancelClick}>Cancel</button>
+  </div>
+);
 
 export default AskForm;
```
Comment (kcelentano) [inline client/components/AskForm.jsx]: Another smart refactor -- only thing, extra line needed.

File: client/components/VoteSwitch.jsx [hunk 3]
```diff
@@ -1,15 +1,32 @@
...
+        </div>
+        <div className="downvote" onClick={handleLoginClick}>
+          <span className="vote-tooltip">Not as helpful</span>
+        </div>
+      </div>
+    );
+  }
+}
 
 export default VoteSwitch;
```
Comment (kcelentano) [inline client/components/VoteSwitch.jsx]: Needs extra line

File: client/components/Headshot.jsx [hunk 4]
```diff
@@ -1,23 +1,10 @@
...
-  }
-}
+const Headshot = ({ username, avatarUrl }) => (
+  <div className="user-clickable">
+    <img className="avatar" src={avatarUrl} />
+    <div className="username">{username}</div>
+  </div>
+);
 
 export default Headshot;
```
Comment (kcelentano) [inline client/components/Headshot.jsx]: Needs extra line

**Comment by jsim0809** [comment]
Yeah! In those cases, I was able to refactor because I learned about the HTML keyword `autofocus`, so I could erase all that code I wrote previously using the weird `createRef` stuff.

**Pull request merged**